include/cantera/thermo/PengRobinson.h

```diff
@@ -0,0 +1,359 @@
+//! @file PengRobinson.h
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#ifndef CT_PENGROBINSON_H
+#define CT_PENGROBINSON_H
+
+#include "MixtureFugacityTP.h"
+#include "cantera/base/Array.h"
+
+namespace Cantera
+{
+/**
+ * Implementation of a multi-species Peng-Robinson equation of state
+ *
+ * @ingroup thermoprops
+ */
+class PengRobinson : public MixtureFugacityTP
+{
+public:
+    //! @name Constructors and Duplicators
+    //! @{
+
+    //! Base constructor.
+    PengRobinson();
+
+    //! Construct and initialize a PengRobinson object directly from an
+    //! ASCII input file
+    /*!
+     * @param infile    Name of the input file containing the phase YAML data
+     *                  to set up the object
+     * @param id        ID of the phase in the input file. Defaults to the empty
+     *     string.
+     */
+    PengRobinson(const std::string& infile, const std::string& id="");
+
+    virtual std::string type() const {
+        return "PengRobinson";
+    }
+
+    //! @name Molar Thermodynamic properties
+    //! @{
+
+    virtual double cp_mole() const;
+    virtual double cv_mole() const;
+
+    //! @}
+    //! @name Mechanical Properties
+    //! @{
+
+    //! Return the thermodynamic pressure (Pa).
+    /*!
+     *  Since the mass density, temperature, and mass fractions are stored,
+     *  this method uses these values to implement the
+     *  mechanical equation of state \f$ P(T, \rho, Y_1, \dots, Y_K) \f$.
+     *
+     * \f[
+     *    P = \frac{RT}{v-b_{mix}} - \frac{\left(\alpha a\right)_{mix}}{v^2 + 2b_{mix}v - b_{mix}^2 }
+     * \f]
+     *
+     *  where:
+     *
+     * \f[
+     *    \alpha = \left[ 1 + \kappa \left(1-T_r^{0.5}\right)\right]^2
+     * \f]
+     *
+     *  and
+     *
+     * \f[
+     *    \kappa = \left(0.37464 + 1.54226\omega - 0.26992\omega^2\right)                              for omega <= 0.491
+     *    \kappa = \left(0.379642 + 1.487503\omega - 0.164423\omega^2 +  0.016667\omega^3 \right)      for omega > 0.491
+     * \f]
+     *
+     *Coefficients a_mix, b_mix and (a \alpha)_{mix} are caclulated as
+     *
+     *\f[
+     *a_{mix} = \sum_i \sum_j X_i X_j a_{i, j} = \sum_i \sum_j X_i X_j sqrt{a_i a_j}
+     *\f]
+     *
+     *\f[
+     *   b_{mix} = \sum_i X_i b_i
+     *\f]
+     *
+     *\f[
+     *   {a \alpha}_{mix} = \sum_i \sum_j X_i X_j {a \alpha}_{i, j} = \sum_i \sum_j X_i X_j sqrt{a_i a_j} sqrt{\alpha_i \alpha_j}
+     *\f]
+     *
+     *
+     */
+    
+    virtual double pressure() const;
+
+    // @}
+
+public:
+
+    //! Returns the standard concentration \f$ C^0_k \f$, which is used to
+    //! normalize the generalized concentration.
+    /*!
+     * This is defined as the concentration by which the generalized
+     * concentration is normalized to produce the activity. In many cases, this
+     * quantity will be the same for all species in a phase. 

```

Comment:
You have a space at the end of the line here and on the next line. This is another editor configuration that you can set, let me know if you want to know how to set it up.

Reply:
Corrected.